Userland/Libraries/LibGfx/ImageFormats/WebPLoaderLossyTables.h

```diff
@@ -11,6 +11,214 @@
 namespace Gfx {
 
 using Prob = u8;
+using tree_index = i8;
+
+// https://datatracker.ietf.org/doc/html/rfc6386#section-10 "Segment-Based Feature Adjustments"
+const tree_index mb_segment_tree[2 * (4 - 1)] = {
+    2, 4,   /* root: "0", "1" subtrees */
+    -0, -1, /* "00" = 0th value, "01" = 1st value */
+    -2, -3  /* "10" = 2nd value, "11" = 3rd value */
+};
+
+// https://datatracker.ietf.org/doc/html/rfc6386#section-8.2 "Tree Coding Example"
+// Repeated in https://datatracker.ietf.org/doc/html/rfc6386#section-11.2 "Luma Modes"
+enum intra_mbmode {
+    DC_PRED,               /* predict DC using row above and column to the left */
+    V_PRED,                /* predict rows using row above */
+    H_PRED,                /* predict columns using column to the left */
+    TM_PRED,               /* propagate second differences a la "True Motion" */
+    B_PRED,                /* each Y subblock is independently predicted */
+    num_uv_modes = B_PRED, /* first four modes apply to chroma */
+    num_ymodes             /* all modes apply to luma */
```

Comment:
is it guaranteed that this will assign `5` to num_ymodes? I guess it's copied directly from the spec but it would be easier to be convinced it's correct if we did `=5` or `=B_PRED+1` here or some such.

Reply:
Sure, enum values are assigned consecutively.